src/main/java/com/thealgorithms/misc/MirrorOfMatrix.java

```diff
@@ -0,0 +1,50 @@
+package com.thealgorithms.misc;
+
+// Problem Statement
+/*
+We have given an array of m x n (where m is the number of rows and n is the number of columns).
+Print the new matrix in such a way that the new matrix is the mirror image of the original matrix.
+
+The Original matrix is:   |   The Mirror matrix is:
+1	2	3	              |   3	 2	 1
+4	5	6	              |   6	 5	 4
+7	8	9	              |   9	 8	 7
+
+@author - Aman (https://github.com/Aman28801)
+*/
+
+public final class MirrorOfMatrix {
+    MirrorOfMatrix() {
+    }
+    public static int[][] mirrorMatrix(int[][] originalMatrix) {
+        if (originalMatrix == null) {
+            // Handle invalid input
+            return null;
+        }
+        if (originalMatrix.length == 0) {
+            return new int[0][0];
+        }
+        int numRows = originalMatrix.length;
+        int numCols = originalMatrix[0].length;
+
+        for (int i = 1; i < numRows; i++) {
+            if (originalMatrix[i].length != numCols) {
+                throw new IllegalArgumentException("Input matrix is not a valid matrix. Rows have different sizes.");
```

Comment:
```suggestion
                throw new IllegalArgumentException("The input is not a matrix.");
```

By the way: this line is not covered by any of the tests. Please have a look here:
https://github.com/TheAlgorithms/Java/blob/536978919dc79af9bf25c57ebb90802d1830784e/src/test/java/com/thealgorithms/maths/NthUglyNumberTest.java#L72